src/components/Carousel.tsx

```diff
@@ -1,18 +1,104 @@
 import React from 'react';
+import cn from 'classnames';
 import './Carousel.scss';
 
-const Carousel: React.FC = () => (
-  <div className="Carousel">
-    <ul className="Carousel__list">
-      <li><img src="./img/1.png" alt="1" /></li>
-      <li><img src="./img/1.png" alt="2" /></li>
-      <li><img src="./img/1.png" alt="3" /></li>
-      <li><img src="./img/1.png" alt="4" /></li>
-    </ul>
-
-    <button type="button">Prev</button>
-    <button type="button">Next</button>
-  </div>
-);
-
-export default Carousel;
+type Props = {
+  images: string[];
+  itemWidth: number;
+  frameSize: number;
+  step: number;
+  animationDuration: number;
+  infinite: boolean;
+  currentImage: number;
+  setCurrentImage: (value: number) => void;
+};
+
+export const Carousel: React.FC<Props> = ({
+  images,
+  itemWidth,
+  frameSize,
+  step,
+  animationDuration,
+  infinite,
+  currentImage,
+  setCurrentImage,
+}) => {
+  const carouselWidth = itemWidth * frameSize;
+  const translateDistance = currentImage * itemWidth;
+  const maxImages = images.length - frameSize;
+
+  function handlePrev() {
```

Comment:
Consider will be better if you use a function expression here
```suggestion
  const handlePrev = () => {
```